services/eth_gateway.go

```diff
@@ -0,0 +1,69 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// TODO: Find etherum node to connect to.
+const ethUrl = "ws://0.0.0.0"
+
+// Singleton client
+var (
+	client *ethclient.Client
+	mtx    sync.Mutex
+)
+
+func sharedClient() (c *ethclient.Client, err error) {
+	if client != nil {
+		return client, nil
+	}
+
+	// check-lock-check pattern to avoid excessive locking.
+	mtx.Lock()
+	defer mtx.Unlock()
+
+	if client != nil {
+		c, err = ethclient.Dial(ethUrl)
+		if err != nil {
+			fmt.Println("Failed to connect to Etherum node.")
+			return
+		}
+
+		// Sets Singleton
+		client = c
+	}
+
+	return client, err
+}
+
+func BuryPrl() {
+	// TODO
+}
+
+// TODO: Don't use floats for money transactions!
+func SendTransaction(fromAddr common.Address, toAddr common.Address, amt float64) {
+	// TODO
+}
+
+// SubscribeToTransfer will subscribe to transfer events
+// sending PRL to the brokerAddr given. Notifications
+// will be sent in the out channel provided.
+func SubscribeToTransfer(brokerAddr common.Address, outCh chan<- types.Log) {
+	ethCl, _ := sharedClient()
+	ctx := context.Background() // TODO: Should we have some timeout or cancel?
+
+	q := ethereum.FilterQuery{
+		FromBlock: nil, // TODO: Figure out how to get current block
```

Comment:
Alex Moyano had to do something similar today for the snapshot script, he probably has some good ideas of how to get the current block

Reply:
@automyr ^

Reply:
More concretely, here's the method using the JSON RPC API:

https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_blocknumber